### ISSUE TITLE
After you stop or close the player with a cross sign and start it again, the song crashes out of the application?

### ISSUE THREAD

**Issue by hajiparvane** [Zeratyll/flutter_background_audio#4]
The following message is displayed :

`unfortunately, background_audio_example has stopped`

and then log : 
```

I/flutter ( 3204): BackgroundAudio: Event "play"
I/flutter ( 3204): BackgroundAudio: Calling listeners for "toggle_playback"
I/flutter ( 3204): BackgroundAudio: Event "stop"
I/flutter ( 3204): BackgroundAudio: Calling listeners for "stop"
D/AndroidRuntime( 3204): Shutting down VM
E/AndroidRuntime( 3204): FATAL EXCEPTION: main
E/AndroidRuntime( 3204): Process: com.backgroundaudioexample, PID: 3204
E/AndroidRuntime( 3204): java.lang.RuntimeException: Unable to create service com.backgroundaudio.AudioPlayer: java.lang.IndexOutOfBoundsException: Invalid index 0, size is 0
E/AndroidRuntime( 3204): 	at android.app.ActivityThread.handleCreateService(ActivityThread.java:2887)
E/AndroidRuntime( 3204): 	at android.app.ActivityThread.-wrap4(ActivityThread.java)
E/AndroidRuntime( 3204): 	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1427)
E/AndroidRuntime( 3204): 	at android.os.Handler.dispatchMessage(Handler.java:102)
E/AndroidRuntime( 3204): 	at android.os.Looper.loop(Looper.java:148)
E/AndroidRuntime( 3204): 	at android.app.ActivityThread.main(ActivityThread.java:5417)
E/AndroidRuntime( 3204): 	at java.lang.reflect.Method.invoke(Native Method)
E/AndroidRuntime( 3204): 	at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:726)
E/AndroidRuntime( 3204): 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:616)
E/AndroidRuntime( 3204): Caused by: java.lang.IndexOutOfBoundsException: Invalid index 0, size is 0
E/AndroidRuntime( 3204): 	at java.util.ArrayList.throwIndexOutOfBoundsException(ArrayList.java:255)
E/AndroidRuntime( 3204): 	at java.util.ArrayList.get(ArrayList.java:308)
E/AndroidRuntime( 3204): 	at com.backgroundaudio.AudioPlayer.getSong(AudioPlayer.java:108)
E/AndroidRuntime( 3204): 	at com.backgroundaudio.AudioPlayer.showNotification(AudioPlayer.java:271)
E/AndroidRuntime( 3204): 	at com.backgroundaudio.AudioPlayer.onCreate(AudioPlayer.java:69)
E/AndroidRuntime( 3204): 	at android.app.ActivityThread.handleCreateService(ActivityThread.java:2877)
E/AndroidRuntime( 3204): 	... 8 more
I/Process ( 3204): Sending signal. PID: 3204 SIG: 9
Lost connection to device.


```

**Comment by N9vn1njdf** [comment]
`Caused by: java.lang.IndexOutOfBoundsException: Invalid index 0, size is 0` 
this error says that the array of songs is empty. This is a plugin bug. I may fix it in the future, now there is no time.

To avoid this error, manually check the array of songs before each action (stop, pause ...)

**Comment by oversoul** [comment]
Fixed on #8
